leader/directed_ring/peterson.go

```diff
@@ -0,0 +1,173 @@
+package directed_ring
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/krzysztof-turowski/distributed-framework/lib"
+	"log"
+)
+
+type statePeterson struct {
+	Max    int
+	Status modeType
+}
+
+type messagePeterson struct {
+	Max  int
+	Mode modeType
+}
+
+func initializePeterson(v lib.Node) bool {
+	s := statePeterson{Max: v.GetIndex(), Status: unknown}
+	data, _ := json.Marshal(s)
+	v.SetState(data)
+	return false
+}
+
+func sendMessagePeterson(v lib.Node, m messagePeterson) {
+	outMessage, _ := json.Marshal(m)
+	v.SendMessage(0, outMessage)
+}
+
+func receiveMessagePeterson(v lib.Node) messagePeterson {
+	var m messagePeterson
+	inMessage := v.ReceiveMessage(0)
+	json.Unmarshal(inMessage, &m)
+	return m
+}
+
+func getStatePeterson(v lib.Node) statePeterson {
+	var s statePeterson
+	json.Unmarshal(v.GetState(), &s)
+	return s
+}
+
+func setStatePeterson(v lib.Node, s statePeterson) {
+	data, _ := json.Marshal(s)
+	v.SetState(data)
+}
+
+func processPeterson(v lib.Node, round int) bool {
+	s := getStatePeterson(v)
+	if s.Status != relay {
+		if s.Status == unknown {
```

Comment:
Please collapse these two ifs into:
if s.Status == unknown {
} else if s.Status == leader {
} else if s.Status == relay {
}

Reply:
thank you